src/test/java/uk/gov/pay/api/resources/directdebit/DirectDebitPaymentsResourceIT.java

```diff
@@ -0,0 +1,289 @@
+                        .build(),
+                someParameters()
+                        .withReference("")
+                        .withErrorMessage("Missing mandatory attribute: reference")
+                        .withErrorField("reference")
+                        .withErrorCode("P0101")
+                        .build(),
+                someParameters()
+                        .withReference("   ")
+                        .withErrorMessage("Missing mandatory attribute: reference")
+                        .withErrorField("reference")
+                        .withErrorCode("P0101")
+                        .build(),
+                someParameters()
+                        .withReference(RandomStringUtils.randomAlphanumeric(256))
+                        .withErrorMessage("Invalid attribute value: reference. Must be less than or equal to 255 characters length")
+                        .withErrorField("reference")
+                        .withErrorCode("P0102")
+                        .build(),
+                someParameters()
+                        .withDescription(RandomStringUtils.randomAlphanumeric(256))
+                        .withErrorMessage("Invalid attribute value: description. Must be less than or equal to 255 characters length")
+                        .withErrorField("description")
+                        .withErrorCode("P0102")
+                        .build(),
+                someParameters()
+                        .withMandateId(null)
+                        .withErrorMessage("Missing mandatory attribute: mandate_id")
+                        .withErrorField("mandate_id")
+                        .withErrorCode("P0101")
+                        .build(),
+                someParameters()
+                        .withMandateId("")
+                        .withErrorMessage("Missing mandatory attribute: mandate_id")
+                        .withErrorField("mandate_id")
+                        .withErrorCode("P0101")
+                        .build(),
+                someParameters()
+                        .withMandateId("   ")
+                        .withErrorMessage("Missing mandatory attribute: mandate_id")
+                        .withErrorField("mandate_id")
+                        .withErrorCode("P0101")
+                        .build(),
+                someParameters()
+                        .withMandateId(RandomStringUtils.randomAlphanumeric(27))
+                        .withErrorMessage("Invalid attribute value: mandate_id. Must be less than or equal to 26 characters length")
+                        .withErrorField("mandate_id")
+                        .withErrorCode("P0102")
+                        .build(),
+        };
+    }
+
+    String paymentEventsLocationFor(String chargeId) {
+        return configuration.getBaseUrl() + "v1/directdebit/payments/" + chargeId + "/events";
+    }
+
+    private String paymentLocationFor(String chargeId) {
```

Comment:
why's this private when the rest aren't?

Reply:
hmm should just be in the base class anyway, i'll update